File: backend/app/schemas/validation.py
from datetime import datetime
from uuid import UUID

from app.models.enums import FindingSeverity
from app.schemas.common import ORMModel


class FindingOut(ORMModel):
    id: UUID
    invoice_id: UUID
    rule_code: str
    severity: FindingSeverity
    message: str
    reference_json: dict | None
    created_at: datetime

File: frontend/src/api/validation.ts
import { api } from "@/api/client";

export type FindingSeverity = "info" | "warning" | "blocker";

export type Finding = {
  id: string;
  invoice_id: string;
  rule_code: string;
  severity: FindingSeverity;
  message: string;
  reference_json: Record<string, unknown> | null;
  created_at: string;
};

export const runValidation = (invoiceId: string): Promise<Finding[]> =>
  api<Finding[]>(`/validation/${invoiceId}/run`, { method: "POST" });

export const listFindings = (invoiceId: string): Promise<Finding[]> =>
  api<Finding[]>(`/validation/${invoiceId}/findings`);
